support-frontend/app/services/PayPalNvpService.scala

```diff
@@ -90,14 +92,48 @@ class PayPalNvpService(apiConfig: PayPalConfig, wsClient: WSClient) extends Touc
       "RETURNURL" -> returnUrl,
       "CANCELURL" -> cancelUrl,
       "BILLINGTYPE" -> "MerchantInitiatedBilling",
-      "NOSHIPPING" -> "1"
+      "NOSHIPPING" -> noShipping
     )
 
     nvpRequest(paymentParams).map { resp =>
       retrieveNVPParam(resp, "TOKEN")
     }
   }
 
+  // Sets up a payment by contacting PayPal and also fetches the users details.
+  def createAgreementAndRetrieveUser(token: Token): Future[Option[PayPalCheckoutDetails]] =
+    createBillingAgreement(token).flatMap(
+      maybeBaid =>
+      retrieveUserInformation(token).map(maybeUserDetails =>
+        maybeBaid match {
+          case Some(baid) => Some(PayPalCheckoutDetails(baid, maybeUserDetails))
+          case _ => None
+        }
+      )
+    )
```

Comment:
is this the same as this:
```suggestion
    for {
      maybeBaid <- createBillingAgreement(token)
      maybeUserDetails <- retrieveUserInformation(token)
    } yield maybeBaid.map(baid => PayPalCheckoutDetails(baid, maybeUserDetails))
```

Reply:
No I don't think it is. In the case where we have a baid but no user information (this will happen for contributions for instance) the first piece of code will return 
`Some(PayPalCheckoutDetails(String, None))`
the second will return `None`.

Reply:
I don't think the second will return None.  It would have if I used an OptionT, but I could see that behaviour wasn't what was implemented.

Reply:
Oh yes I see what you're saying, sorry I had OptionT in my head.